extends Area3D

signal ya_got_hit

@export var time_to_despawn_in_seconds = 2
@onready var timer = $Timer


var velocity = Vector3.ZERO
# Called when the node enters the scene tree for the first time.
func _ready():
	timer.wait_time = time_to_despawn_in_seconds
	timer.start()
	pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _physics_process(delta):
	global_transform.origin  += velocity * delta
	pass



func _on_body_entered(body):
	if body.is_in_group("Player"):
		emit_signal("ya_got_hit")
		#print("ya_got_hit")
	elif body is StaticBody3D:
		$".".queue_free()
	pass # Replace with function body.




func _on_timer_timeout():
	$".".queue_free()
	print("freed!!!!" + str(timer.wait_time))
	pass # Replace with function body.
